tests/cxx/placement_new.cc

```diff
@@ -0,0 +1,152 @@
+//===--- placement_new.cc - test input file for iwyu ----------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// Test that use of placement-new requires include of <new> in all the usual
+// scenarios.
+//
+// Requires -std=c++17 on the command-line to have std::aligned_val_t available
+// in <new>.
+
+#include "tests/cxx/direct.h"
+#include "tests/cxx/placement_new-d1.h"
+
+// Placement-new of builtin types.
+void PlacementNewBuiltinType() {
+  // Make sure we only report a use of <new> because of placement-new, not
+  // ordinary new-expressions.
+  int* newed_int = new int;
+  // IWYU: operator new is...*<new>
+  new (newed_int) int(4);
+
+  delete newed_int;
+}
+
+// Placement-new of user-defined type.
+void PlacementNewUserType() {
+  // IWYU: IndirectClass needs a declaration
+  // IWYU: IndirectClass is...*indirect.h
+  IndirectClass* icptr = new IndirectClass;
```

Comment:
Need to delete this at the end of the function.

Reply:
Thanks, done. On another branch, coming soon.